### PR TITLE
mesa: build Intel drivers on i686

### PR DESCRIPTION
**Pull request by K900**
Sorry gamers :(

###### Description of changes

Pointed out by @m911t at https://github.com/NixOS/nixpkgs/pull/213158#issuecomment-1416309016

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
Another one for the Tuesday list, I suppose.

**Comment by m911t** [comment]
in the meantime, will an overlay that sets galliumDrivers to just "iris" fix this on uh a local level?

**Comment by K900** [comment]
Yes.

**Comment by m911t** [comment]
wowee!

**Comment by vcunat** [comment]
Or use a slightly older nixpkgs commit to have all these cached in binary form (though yes, some security fixes happened in between).

File: pkgs/development/libraries/mesa/default.nix [hunk 1]
```diff
@@ -28,9 +28,9 @@
       "tegra" # Nvidia Tegra SoCs
       "v3d" # Broadcom VC5 (Raspberry Pi 4)
       "vc4" # Broadcom VC4 (Raspberry Pi 0-3)
-    ] ++ lib.optionals stdenv.isx86_64 [
-      "iris" # new Intel, could work on non-x86_64 with PCIe cards, but doesn't build as of 22.3.4
-      "crocus" # Intel legacy, x86_64 only
+    ] ++ lib.optionals stdenv.isx86 [
```
Comment (trofi) [inline pkgs/development/libraries/mesa/default.nix:31]: Fails eval. `stdenv.hostPlatform.isx86` perhaps?

Comment (wegank) [reply pkgs/development/libraries/mesa/default.nix:31]: Maybe `stdenv.isx86_32`?

Comment (vcunat) [reply pkgs/development/libraries/mesa/default.nix:31]: `stdenv.hostPlatform.isx86` should be easiest, as we want both bit-ness.

**Comment by K900** [comment]
Fixed.

**Review by trofi — APPROVED** [review]

**Review by PedroHLC — APPROVED** [review]
LGTM

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by m911t** [comment]
There's still no iris_dri.so on /run/opengl-driver-32/lib/dri/ but it is there on the 64bit folder. I can't use anything 32bit still.

**Comment by K900** [comment]
What version of nixpkgs are you on?

**Comment by vcunat** [comment]
This got into `nixos-unstable` just a couple hours ago, and I do se the file in there (symlink to `crocus_dri.so`).

**Comment by m911t** [comment]
Huh, weird, i just did a nixos-rebuild --upgrade-all and it's not there.

**Comment by K900** [comment]
What channel are you on?

**Comment by m911t** [comment]
nixos-unstable, I literally did an upgrade an hour ago, doing it again and I'm getting new stuff.

**Comment by vcunat** [comment]
You might've missed it.  Check commit.  It would be 545c7a31e5d to get it.

**Comment by m911t** [comment]
How can I do that? Sorry for being annoying..

**Comment by vcunat** [comment]
If it's running already, I'd use `nixos-version` command.  Or for standard channels
```bash
cat /nix/var/nix/profiles/per-user/root/channels/nixos/.git-revision
```

**Comment by m911t** [comment]
ah, got it now!
Thank you so much, sorry for being too anxious.

**Pull request merged**